src/tlo/methods/healthsystem.py

```diff
@@ -2599,6 +2715,22 @@ def apply(self, population):
                     priority=-1,
                     did_run=True,
                 )
+                if len(_inpatient_appts) != 0:
+                    self.module.write_to_hsi_no_blank_footprint_log(
```

Comment:
to avoid this long (and mostly repetitive call), I think it would be neater to let the logic be inside the existing function (write_to_hsi_log), That way, we have one place where we're doing the split between blank and non-blank footprints.